gensim/similarities/annoy.py

```diff
@@ -156,6 +156,7 @@ def build_from_word2vec(self):
 
     def build_from_doc2vec(self):
         """Build an Annoy index using document vectors from a Doc2Vec model."""
+        # FIXME docvecs now = ? What's the replacement for docvecs.count, index_to_doctag, etc?
         docvecs = self.model.docvecs
         labels = [docvecs.index_to_doctag(i) for i in range(docvecs.count)]
         return self._build_from_model(docvecs.vectors_docs_norm, labels, self.model.vector_size)
```

Comment:
There's no `doc2VecKeyedVectors.vectors_docs_norm` anymore, either as a doc-vectors-specific set of norms, nor as an usually-realized large array of all normed vectors. Instead, options include: (1) requesting normed vectors one-by-one with `dv.get_vector(key, norm=True)`; (2) `dv.get_normed_vectors()` which actually calcs the whole dense array on demand (without caching it for future requests) if truly needed.

I suspect `_build_from_model()`, below, may not be as important as `build_from_keyed_vectors()`. If it exists and needs the normed vectors (& I'm not sure it should force norming on all users, sometimes euclidean distances on raw vectors win on certain tasks!), it could unit-norm them itself, or let the user know they should pre-norm if that's what they know they want.

Reply:
Also re: `build_from_keyedvectors()` below - it may want to avoid instantiating the whole `get_normed_vectors()` - to use much less peak memory, but perhaps at cost of slower looping over each individually. (Whether that makes any different given the one-at-a-time `add_item()` further below, unsure.) I fear this class may in many cases be an 'attractive nuisance', which people assume is a good/necessary improvement since it's there, and we have a demo, but often isn't worth the overhead/complexity. See, for example, recent memory issue #2883 resolved satisfactorily just by dropping the complexifying use of AnnoyIndexer.

Reply:
Yes, already fixed.